processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -407,35 +216,56 @@ public T get(int index)
   @Override
   public int indexOf(T value)
   {
-    return bufferIndexed.indexOf(value);
+    return indexOf(this, value);
+  }
+
+  private int indexOf(Indexed<T> indexed, T value)
```

Comment:
Can you please elaborate a bit more on which number is which for the benchmark?  All I see are two numbers that are likely equivalent and just differ based on when the thing was run.

Also, can you add benchmarks to
1. verify that the performance of indexOf() isn't adversely affected
2. verify which works better when both v1 and v2 are active?